### PR TITLE
adding TLS config for Go

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?
Adds TLS config example for Go to the app-dev guide.

## Notes to reviewers
Removed Python link to PR.

### CODE REVIEW

File: docs/application-development/foundations.md [hunk 1]
```diff
@@ -625,7 +625,21 @@ values={[{label: 'Go', value: 'go'},{label: 'Java', value: 'java'},{label: 'PHP'
 
 <TabItem value="go">
 
-Content is not available
+Use [`ConnectionOptions`](https://pkg.go.dev/go.temporal.io/sdk@v1.15.0/client#ConnectionOptions) API available in the [`go.temporal.io/sdk/client`](https://pkg.go.dev/go.temporal.io/sdk/client) package to connect a Client with TLS.
+
+```go
+	cert, err := tls.LoadX509KeyPair(clientCertPath, clientKeyPath)
```
Comment (bergundy) [inline docs/application-development/foundations.md:631]: Eventually we'll need to snipsync this from the samples but this is fine for now.

File: docs/application-development/foundations.md [hunk 2]
```diff
@@ -625,7 +625,21 @@ values={[{label: 'Go', value: 'go'},{label: 'Java', value: 'java'},{label: 'PHP'
 
 <TabItem value="go">
 
-Content is not available
+Use [`ConnectionOptions`](https://pkg.go.dev/go.temporal.io/sdk@v1.15.0/client#ConnectionOptions) API available in the [`go.temporal.io/sdk/client`](https://pkg.go.dev/go.temporal.io/sdk/client) package to connect a Client with TLS.
```
Comment (bergundy) [inline docs/application-development/foundations.md:628]: Is this rendered content? Pardon my ignorance, it's the first time I'm reviewing the app dev guide.

File: docs/go/how-to-set-tls-configuration-in-go.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+title: How to set TLS configuration in Go
+sidebar_label: Set TLS configuration
+description: Set TLS configuration
+tags:
+  - developer-guide
+  - sdk
+  - Go
+---
+
+Use [`ConnectionOptions`](https://pkg.go.dev/go.temporal.io/sdk@v1.15.0/client#ConnectionOptions) API available in the [`go.temporal.io/sdk/client`](https://pkg.go.dev/go.temporal.io/sdk/client) package to connect a Client with TLS.
```
Comment (bergundy) [inline docs/go/how-to-set-tls-configuration-in-go.md:12]: Do we want to link to a specific version?
Do we make sure to update all the links when a new version is released?

File: docs/go/how-to-set-tls-configuration-in-go.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+id: how-to-set-tls-configuration-in-go
+title: How to set TLS configuration in Go
+sidebar_label: Set TLS configuration
+description: Set TLS configuration
```
Comment (bergundy) [inline docs/go/how-to-set-tls-configuration-in-go.md:5]: I'd clarify that this is for mTLS, for server side TLS only there are different options.

File: docs/go/how-to-set-tls-configuration-in-go.md (hunk 3 above)
Comment (lorensr) [inline docs/go/how-to-set-tls-configuration-in-go.md:12]:
```suggestion
Use [`ConnectionOptions`](https://pkg.go.dev/go.temporal.io/sdk/client#ConnectionOptions) API available in the [`go.temporal.io/sdk/client`](https://pkg.go.dev/go.temporal.io/sdk/client) package to connect a Client with TLS.
```

Comment (bergundy) [reply docs/go/how-to-set-tls-configuration-in-go.md]: I see, it's just that you linked to a specific version here, so I asked.

**Review by cretz — APPROVED** [review]

**Review by djmagee — APPROVED** [review]

**Pull request merged**